### PR TITLE
ASC-1204 Fix test_get_nova_allocation_ratios

### PR DESCRIPTION
**Pull request by johnduarte**
This commit fixes an error in connecting to the `nova_conductor` host.
The host is now determined by introspecting the intersection of hosts in
the `nova_conductor` group that are also in the `infraX-host_containers`
group for the infraX target host.

### CODE REVIEW

File: molecule/default/tests/test_hypervisor_free.py [hunk 1]
```diff
@@ -25,8 +26,17 @@ def test_get_nova_allocation_ratios(host):
     global ram_ratio
     global disk_ratio
```
Comment (ryan-rs) [inline molecule/default/tests/test_hypervisor_free.py]: Eek! Global is definitely not recommended in Python if you can avoid it.

File: molecule/default/tests/test_hypervisor_free.py [hunk 2]
```diff
@@ -5,16 +5,17 @@
...
 """ASC-157: Perform Post Deploy System validations"""
 
+target_host = 'infra1'
 testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
-    os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('infra1')
+    os.environ['MOLECULE_INVENTORY_FILE']).get_hosts(target_host)
+
 cpu_ratio = 1
 ram_ratio = 1
 disk_ratio = 1
```
Comment (phongdly) [inline molecule/default/tests/test_hypervisor_free.py]: Can the *_ratio be defined in a fixture to avoid global variable below?

**Comment by johnduarte** [comment]
@ryan-rs @phongdly , ratio lookup converted to a fixture.

File: molecule/default/tests/test_hypervisor_free.py [hunk 3]
```diff
@@ -5,28 +5,35 @@
...
-ram_ratio = 1
-disk_ratio = 1
+    os.environ['MOLECULE_INVENTORY_FILE']).get_hosts(target_host)
 
 
 @pytest.mark.test_id('d7fc4ff0-432a-11e8-aca5-6a00035510c0')
-@pytest.mark.jira('asc-157')
-@pytest.mark.skip(reason='expected ratio values not defined')
-def test_get_nova_allocation_ratios(host):
+@pytest.mark.jira('asc-157', 'asc-1204')
```
Comment (ryan-rs) [inline molecule/default/tests/test_hypervisor_free.py]: Fixtures don't need 'jira' or 'test_id' marks.

**Comment by johnduarte** [comment]
@ryan-rs extraneous marks have been removed from the fixture.

**Pull request merged**